core/src/main/scala/za/co/absa/spline/core/SparkLineageInitializer.scala

```diff
@@ -58,31 +69,56 @@ object SparkLineageInitializer extends Logging {
       * @return An original Spark session
       */
     def enableLineageTracking(configurer: SplineConfigurer = defaultSplineConfigurer): SparkSession = {
-      if (configurer.splineMode != DISABLED) sparkSession.synchronized {
-        preventDoubleInitialization()
-        log info s"Spline v${SplineBuildInfo.version} is initializing..."
-        try {
-          attemptInitialization(configurer)
-          log info s"Successfully initialized. Spark Lineage tracking is ENABLED."
-        } catch {
-          case NonFatal(e) if configurer.splineMode == BEST_EFFORT =>
-            log.error(s"Initialization failed! Spark Lineage tracking is DISABLED.", e)
+      val splineConfiguredForCodelessInit = sparkSession.sparkContext.getConf
+        .getOption(QUERY_EXECUTION_LISTENERS.key).toSeq
+        .flatMap(s => s.split(",").toSeq)
+        .contains(classOf[CodelessQueryExecutionListener].getCanonicalName)
+      if (!splineConfiguredForCodelessInit || spark.SPARK_VERSION.startsWith("2.2")) {
+        if (splineConfiguredForCodelessInit) {
+          log.warn("Spline lineage tracking is also configured for codeless initialization, but codeless init is supported on Spark 2.3+ and not current version 2.2. Spline will be initialized only via code call to enableLineageTracking i.e. the same way as is now.")
         }
+        createQueryExecutionListener(configurer)
+          .foreach(sparkSession.listenerManager.register(_))
+        // TODO: SL-128
```

Comment:
Please remove all TODOs. According to the agreement all TODOs should be either resolved in the current PR or a new issue should be created on GitHub. In exceptional cases, when a TODO cannot be fixed now but really helps readability (or, to be more precise, absence of it would hurt the code readability) then it can be left with the bi-directional reference to the respective github issue.

Reply:
I removed this, but it also comes from previous code.